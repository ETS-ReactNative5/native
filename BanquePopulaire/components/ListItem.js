import React from 'react';
import { StyleSheet, View, Image, Text, TouchableOpacity } from "react-native";

export default class ListItem extends React.Component {
    renderChildren() {
        const childrenArray = React.Children.map(
            this.props.children,
            child => child
        );

        return childrenArray;
    }

    render() {
        return (
            <TouchableOpacity onPress={this.props.onPress}>
                <View style={[this.props.style, styles.container]}>
                    {this.renderChildren()}
                </View>
            </TouchableOpacity>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'row',
    }
})