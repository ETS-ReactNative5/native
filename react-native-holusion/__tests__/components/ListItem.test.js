import React from 'react'
import {Text} from 'react-native'

import ListItem from '../../src/components/ListItem'

import renderer from 'react-test-renderer'

test('renders ListItem correctly', () => {
  let tree = renderer.create(<ListItem><Text>Blabla</Text></ListItem>).toJSON();
  expect(tree).toMatchSnapshot();

  tree = renderer.create(<ListItem><Text>Foo</Text><Text>Bar</Text></ListItem>).toJSON();
  expect(tree).toMatchSnapshot();
})
