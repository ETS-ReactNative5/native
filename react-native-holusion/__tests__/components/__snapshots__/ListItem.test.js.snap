// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListItem correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Blabla
    </Text>
  </View>
</View>
`;

exports[`renders ListItem correctly 2`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Bar
    </Text>
  </View>
</View>
`;
